variables.tf

```diff
@@ -240,8 +240,14 @@ variable "vpc_config" {
   })
 }
 
-variable "custom_iam_policy_arns" {
-  type        = set(string)
-  description = "List of custom policies to be attached to the lambda role"
-  default     = []
+
+variable "custom_iam_policy_enabled" {
+  type        = string
+  description = "To enable custom policies on the iam role"
+  default     = false
+}
+variable "custom_iam_policy_arn" {
+  type        = string
+  description = "ARN of custom policy to be attached to the lambda role"
+  default     = null
```

Comment:
We ran into this issue before and `length()` for a single policy arn had fixed it so I'm unsure why we would need a `custom_iam_policy_enabled` input

Reply:
If we're creating a new `enabled` arg, why change the `custom_iam_policy_arns` to a single policy ?

Reply:
but if the policy is being created at the same time as the module the length() can't be calculated for the ARN because the policy will be created in the apply stage so it can't be computed, is that the right assumption?